### PR TITLE
Improve RepositoryHelper.getSharedBundlePools() to avoid exceptions

### PR DESCRIPTION
**Pull request by merks**
Just log an exception and return an empty list in the unlikely event of failure.

https://github.com/eclipse-equinox/p2/issues/373

### CODE REVIEW

**Comment by merks** [comment]
Hopefully this is the last round!  😱

**Comment by merks** [comment]
FYI, I've already committed these changes to Oomph and they are in a nightly build:

https://github.com/eclipse-oomph/oomph/issues/41

So soon the encoding.info file will exist everywhere...

File: bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/helpers/RepositoryHelper.java [hunk 1]
```diff
@@ -149,18 +148,19 @@ public static <T> IRepository<T> createMemoryComposite(IProvisioningAgent agent,
...
-					return lines.map(Path::of).filter(Files::isDirectory).toList();
-				}
+	public static List<Path> getSharedBundlePools() {
+		Path bundlePools = Path.of(System.getProperty("user.home"), ".p2/pools.info");
+		if (Files.isRegularFile(bundlePools)) {
+			try (Stream<String> lines = Files.lines(bundlePools, getSharedBundlePoolEncoding())) {
+				return lines.map(Path::of).filter(Files::isDirectory).toList();
+			} catch (Exception ex) {
+				LogHelper.log(
+						new Status(IStatus.WARNING, Activator.ID, "The bundle pool load failed: " + bundlePools, ex));
```
Comment (HannesWell) [inline bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/helpers/RepositoryHelper.java:160]:
```suggestion
						Status.warning("The bundle pool load failed: " + bundlePools, ex));
```
would be a bit simpler. :)

**Review by HannesWell — APPROVED** [review]
LGTM, just one little suggestion. :)

File: bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/helpers/RepositoryHelper.java (hunk 1 above)
Comment (laeubi) [reply bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/helpers/RepositoryHelper.java:160]: `ILog.error("The bundle pool load failed: " + bundlePools, ex);` would even be simpler ;-)

**Review by laeubi — APPROVED** [review]
Makes sense as the user most likely can't do anything than log/ignore the exception if it will ever arise...

File: bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/helpers/RepositoryHelper.java (hunk 1 above)
Comment (HannesWell) [reply bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/helpers/RepositoryHelper.java]: > `ILog.warn("The bundle pool load failed: " + bundlePools, ex);` would even be simpler ;-)

I think it needs to be `ILog.get().warn(...`, but I agree that there is a lot of potential in P2 to improve logging (and other things).^^

Comment (merks) [reply bundles/org.eclipse.equinox.p2.repository/src/org/eclipse/equinox/internal/p2/repository/helpers/RepositoryHelper.java]: What was the comment?  Perfect is the enemy of good.  😈

**Pull request merged**